smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -714,6 +721,26 @@ public boolean isSupported() throws NoResponseException, XMPPErrorException, Not
         return serviceDiscoveryManager.supportsFeature(archiveAddress, MamElements.NAMESPACE);
     }
 
+    public boolean isAdvancedConfigurationSupported(Jid jid) throws InterruptedException, XMPPException, SmackException {
+        DiscoverItems discoverItems = adHocCommandManager.discoverCommands(jid);
+        for (DiscoverItems.Item item : discoverItems.getItems()) {
+            if (item.getNode().equals(ADVANCED_CONFIG_NAMESPACE)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public List<RemoteCommand> getAdvancedConfigurationCommands(Jid jid) throws InterruptedException, XMPPException, SmackException {
+        List<RemoteCommand> remoteCommands = new ArrayList<>();
```

Comment:
This should throw a `FeatureNotSupportedException` if the feature is not supported (instead of returning an empty list).